src/net/tls.cc

```diff
@@ -995,6 +996,9 @@ class session : public enable_lw_shared_from_this<session> {
                     gnutls_certificate_server_set_request(*this, GNUTLS_CERT_REQUEST);
                     break;
                 case client_auth::REQUIRE:
```

Comment:
do we also need to do this for `REQUEST`? based on a very quick glance at the documentation it sounded like with `REQUEST` the handshake might happen, but isn't required?